controllers/nginx/pkg/template/configmap.go

```diff
@@ -87,6 +87,14 @@ func ReadConfig(src map[string]string) config.Configuration {
 		glog.Warningf("unexpected error merging defaults: %v", err)
 	}
 
+	if len(to.LoadBalanceAlgorithm) == 0 {
+		to.LoadBalanceAlgorithm = "least_conn;"
+	} else if to.LoadBalanceAlgorithm == "round_robin" {
+		to.LoadBalanceAlgorithm = ""
+	} else if !strings.HasSuffix(to.LoadBalanceAlgorithm, ";") {
+		to.LoadBalanceAlgorithm += ";"
```

Comment:
Should we assume no `;` and add it in the template?  That way `LoadBalanceAlgorithm` is the algorithm instead of the algorithm + the semi-colon nginx config requires.